buildtest/config.py

```diff
@@ -3,42 +3,48 @@
 import shutil
 from jsonschema import validate
 
-from buildtest.utils.file import create_dir
+from buildtest.buildsystem.schemas.utils import load_schema
 from buildtest.defaults import (
     BUILDTEST_SETTINGS_FILE,
     BUILDTEST_ROOT,
     DEFAULT_SETTINGS_FILE,
     DEFAULT_SETTINGS_SCHEMA,
 )
-from buildtest.buildsystem.schemas.utils import load_schema
+from buildtest.utils.file import create_dir, is_dir, is_file
 
+logger = logging.getLogger(__name__)
 
-def create_settings_file():
+
+def create_settings_file(settings_file):
     """If default settings file don't exist, copy the default settings provided by buildtest."""
 
-    if not os.path.exists(BUILDTEST_SETTINGS_FILE):
-        shutil.copy(DEFAULT_SETTINGS_FILE, BUILDTEST_SETTINGS_FILE)
+    if not is_file(settings_file):
+        logger.debug(f"Detected File: {settings_file} is not present")
+        shutil.copy(DEFAULT_SETTINGS_FILE, settings_file)
+        logger.debug(f"Copying file: {DEFAULT_SETTINGS_FILE} to {settings_file}")
 
 
-def init():
+def init(settings_path):
     """Buildtest init should check that the buildtest user root exists,
        and that dependency files are created. This is called by 
-       ``load_settings``."""
+       ``load_settings``.
+    """
+
+    # if settings_path is not defined dirname is equivalent to BUILDTEST_ROOT, since BUILDTEST_SETTINGS_FILE is in BUILDTEST_ROOT
+    dirname = os.path.dirname(settings_path)
 
     # check if $HOME/.buildtest exists, if not create directory
-    if not os.path.exists(BUILDTEST_ROOT):
-        print(
-            f"Creating buildtest configuration directory: \
-                 {BUILDTEST_ROOT}"
-        )
-        os.mkdir(BUILDTEST_ROOT)
+    if not is_dir(dirname):
+        create_dir(dirname)
+        msg = f"Creating buildtest settings directory: {dirname}"
```

Comment:
This seems redundant - if you want to print a message to the user and logger, use logger.info.

Reply:
okay i can fix this